### PR TITLE
Feat/rental rules

### PR DESCRIPTION
**Pull request by Arseniia-Damaksina**
## Describe your changes

- I added markup for Renral Rules page;
- also added rental rules popup which appears if the user adds the product to cart;
- implemented the logic not to show the rental rules popup if the appropriate checkbox is checked;
- added categories, brand, file and material info to the single product screen;
- fixed styling issues;
- added evental rent to the radio input if the product is in evental category;

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-75)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [x] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.
screem1
screen2
screen3

### CODE REVIEW

File: src/pages/ProductPage/hooks/useProductSection.ts [hunk 1]
```diff
@@ -103,15 +109,27 @@ const useProductSection = (product: IProduct) => {
     }
   };
 
+  const handleAddToCartOrOpenModal = async () => {
+    if (willHideRentalRules) {
+      await handleAddToCart();
+    } else {
+      handleOpen();
+    }
+  };
```
Comment (lyudad) [inline src/pages/ProductPage/hooks/useProductSection.ts]: useCallback

File: src/pages/ProductPage/hooks/useRentalRulesPopup.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  const handleExpandClick = (id: number) => {
+    setExpanded(expanded === id ? null : id);
+  };
+
+  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setIsCheckboxChecked(event.target.checked);
+  };
+
+  const handleAddToCartAndCloseModal = async () => {
```
Comment (lyudad) [inline src/pages/ProductPage/hooks/useRentalRulesPopup.ts]: useCallback

File: src/pages/ProductPage/hooks/useProductSection.ts (hunk 1 above)
Comment (Arseniia-Damaksina) [reply src/pages/ProductPage/hooks/useProductSection.ts]: fixed

File: src/pages/ProductPage/hooks/useRentalRulesPopup.ts (hunk 2 above)
Comment (Arseniia-Damaksina) [reply src/pages/ProductPage/hooks/useRentalRulesPopup.ts]: fixed

**Pull request merged**